src/amplitude-client.js

```diff
@@ -1428,32 +1436,26 @@ var _removeEvents = function _removeEvents(scope, eventQueue, maxId) {
  * Send unsent events. Note: this is called automatically after events are logged if option batchEvents is false.
  * If batchEvents is true, then events are only sent when batch criterias are met.
  * @private
- * @param {Amplitude~eventCallback} callback - (optional) callback to run after events are sent.
- * Note the server response code and response body are passed to the callback as input arguments.
  */
-AmplitudeClient.prototype.sendEvents = function sendEvents(callback) {
+AmplitudeClient.prototype.sendEvents = function sendEvents() {
   if (!this._apiKeySet('sendEvents()')) {
-    if (type(callback) === 'function') {
-      callback(0, 'No request sent', {reason: 'API key not set'});
-    }
+    this.removeEvents(Infinity, Infinity, 0, 'No request sent', {reason: 'API key not set'});
```

Comment:
does this work? maybe i'm mistaken, but doesn't this break on line 1421 above?
where `scope = eventQueue = Infinity` and `scope[eventQueue]` will be `undefined`

Reply:
It's confusing. There is a `removeEvents` function bound to `this` and there's a `_removeEvents` function. I'm calling the former which invokes the latter with `scope`.